### PR TITLE
[Testing] [go-gen] Print go directory structure on Go compilation error

### PR DESCRIPTION
**Pull request by c-harding**
This makes dealing with Go compilation errors slightly easier. The line numbers are printed with the source code.

Also simplifies the logic for encoding `Files` to JSON, using circe’s KeyEncoder

### CODE REVIEW

File: src/it/scala/temple/containers/GolangSpec.scala [hunk 1]
```diff
@@ -25,12 +26,21 @@ abstract class GolangSpec extends DockerShell2HttpService(8081) with DockerTestK
...
       .params(Map("src" -> json, "root" -> entryFile.folder, "entrypoint" -> entryFile.filename))
-      .timeout(connTimeoutMs = 1000, readTimeoutMs = 30000)
+      .timeout(connTimeoutMs = 1000, readTimeoutMs = 60000)
       .asString
       .body
+    if (errors.nonEmpty) {
+      for ((file, content) <- files.to(SortedMap)) {
+        System.err.println(s"$file:")
+        for ((line, i) <- content.linesIterator.zipWithIndex) {
+          System.err.println(s"${i + 1}:".padTo(6, ' ') + line)
```
Comment (jaylees14) [inline src/it/scala/temple/containers/GolangSpec.scala]: 6? 🧙

**Review by jaylees14 — APPROVED** [review]
Nice

File: src/it/scala/temple/containers/GolangSpec.scala (hunk 1 above)
Comment (c-harding) [reply src/it/scala/temple/containers/GolangSpec.scala]: surely no more than 4 digits for the row number, usually 3, then a colon, then a space. Would you rather I commented it?

Comment (jaylees14) [reply src/it/scala/temple/containers/GolangSpec.scala]: I don’t think I would have guessed that explanation, so yes please 👍

**Review by jaylees14 — APPROVED** [review]

**Review by Lewiky — APPROVED** [review]
Nice one

**Pull request merged**